vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -83,6 +88,65 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+func IsErrEncryptedKey(err error) bool {
+	return err != nil && strings.Contains(err.Error(), encryptedPrivateKeyMessage)
+}
```

Comment:
@endophage I think the goal is to get the implementation stubbed out, and then port the changes to `go-connections`:  

> This is a first attempt at fixing #30935. As discussed on #30935, the first draft of the PR includes changes to vendor repositories (docker/go-connections and docker/notary), to allow review of all the required changes. These will be submitted as PRs on corresponding repos after the approval of
the changes.